### PR TITLE
Adds support to add/delete approval rules and get the approval settings

### PR DESCRIPTION
**Pull request by AngelVlc**
I've added these methods:

Related to approval rules:
- delete_approval_rule
- create_approval_rule

And related to approval settings:
- approval_settings_for_project

Thanks!

### CODE REVIEW

File: lib/gitlab/client/approval_rules.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'json'
```
Comment (NARKOZ) [inline lib/gitlab/client/approval_rules.rb]: why is it required?

File: lib/gitlab/client/approval_rules.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    end
+
+    # Crates an approval rule with the name and settings provided
+    #
+    # @example
+    #   Gitlab.create_approval_rule(1, "Default",  { :approvals_required => 2, :users =>[], :groups => [1, 2], :remove_hidden_groups => false })
+    #   Gitlab.delete_approval_rule("project", "Default", { :approvals_required => 2, :users =>[], :groups => [1, 2], :remove_hidden_groups => false })
+    #
+    # @param  [Integer, String] id The ID or name of a project.
+    # @param  [Integer, String] id The ID of the rule to be deleted.
```
Comment (NARKOZ) [inline lib/gitlab/client/approval_rules.rb]: ✂️

File: lib/gitlab/client/approval_rules.rb (hunk 1 above)
Comment (AngelVlc) [reply lib/gitlab/client/approval_rules.rb]: It isn't required, I've removed.

File: lib/gitlab/client/approval_rules.rb (hunk 2 above)
Comment (AngelVlc) [reply lib/gitlab/client/approval_rules.rb]: Fixed.

File: lib/gitlab/client/approval_rules.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class Gitlab::Client
+  # Defines methods related to project approval rules.
+  # Does not exist in API documentation
```
Comment (NARKOZ) [inline lib/gitlab/client/approval_rules.rb]: How did you write the code if there are no docs?

Comment (AngelVlc) [reply lib/gitlab/client/approval_rules.rb]: I didn't write the code but my mates told me that they found them by inspecting the calls from the UI with Chrome 😅

**Comment by NARKOZ** [comment]
I'm not sure that the client should support undocumented (unofficial) API.

**Comment by NARKOZ** [comment]
Sorry, I'm closing this. This client wouldn't support unofficial API.

**Comment by AngelVlc** [comment]
👍

**Comment by johnbatty** [comment]
I was just looking at implementing...
https://docs.gitlab.com/12.8/ee/api/merge_request_approvals.html#get-project-level-rules
Which looks like (part of) the function this PR is trying to implement.
Would it be best to tweak this PR to use this API?

Fyi, I want support for this "project-level-rules" API as I have been trying to use the existing client API [project_merge_request_approvals](https://www.rubydoc.info/gems/gitlab/Gitlab/Client/MergeRequestApprovals#project_merge_request_approvals-instance_method) that calls "api/v4/projects/<id>/approvals) that calls "api/v4/projects/<id>/approvals", and found that the server has a bug that returns incorrect information (I've got an open GitLab issue for this).  Whereas the "get-project-level-rules" works, but this is not currently exposed via the Ruby client library.

**Pull request closed without merging**